main.c

```diff
@@ -215,6 +216,12 @@ static interface_ref start(VDECONN *vdeconn, struct cli_options *cliopt) {
   return iface;
 }
 
+static sigjmp_buf jmpbuf;
+static void signalhandler(int signal) {
+  printf("\nReceived signal %d\n", signal);
```

Comment:
printf is not safe to be used in a signal handler
https://stackoverflow.com/questions/16891019/how-to-avoid-using-printf-in-a-signal-handler

Reply:
It is, but then we would have to call `_exit` instead and rely on the OS to close handles and and other cleanup because we are going to call `siglongjmp` next.

> *  If a signal handler interrupts the execution of an unsafe function, and the handler terminates via a call to [longjmp(3)](https://man7.org/linux/man-pages/man3/longjmp.3.html) or [siglongjmp(3)](https://man7.org/linux/man-pages/man3/siglongjmp.3.html) and the program subsequently calls an unsafe function, then the behavior of the program is undefined.

I don't really care strongly either way: most of the time things work correctly because the signal didn't interrupt an unsafe function, and we get the information printed, and regular cleanup called.

I haven't actually seen it fail at all, but even if it does, we would just see the process crash instead, which would then continue with OS-level cleanup the same way as if we had called `_exit`; but we get a different exit code instead.

Now, given that in non-debugging situations this all runs as a daemon, nobody will see either the output on stdout, nor the crash exit code, so I don't think this matters very much at all.

I'm fine with just calling `_exit(0)` in the signal handler and not doing anything else, if you prefer that.